addons/l10n_jo/models/ir_attachment.py

```diff
@@ -0,0 +1,41 @@
+from odoo import _, api, models
+from odoo.exceptions import UserError
+from odoo.tools.mimetypes import guess_mimetype
+
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    @api.ondelete(at_uninstall=True)
+    def _except_detached_invoice_pdfs(self):
+        detached_pdf_reports = self.filtered(
+            lambda attachment:
+            attachment.res_model == 'account.move'
+            and attachment.res_id
+            and not attachment.res_field
+            and attachment.raw
+            and guess_mimetype(attachment.raw) == 'application/pdf'
+            and attachment.company_id.account_fiscal_country_id.code == 'JO'
+            and 'detached by' in attachment.name
```

Comment:
This is so ugly.  What is detached by is translated.  What if it was translated differently in the attachment than here?